sympy/solvers/solvers.py

```diff
@@ -2667,6 +2668,11 @@ def _tsolve(eq, sym, **flags):
                 return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
             elif lhs.base == 0 and rhs == 1:
                 return _solve(lhs.exp, sym, **flags)
+            elif rhs==1 and lhs.base!=0 :
```

Comment:
this looks too specific; consider the following case where a solution also exists.
```
>>> ((x)**(x+4)-4).subs(x,-2)
0
>>> solve((x)**(x+4)-4)
No algorithms...
```

Reply:
should this be an addition to `diophantine`?

Reply:
```python
        elif lhs.is_Pow:
            if lhs.exp.is_Integer:
                if lhs - rhs != eq:
                    return _solve(lhs - rhs, sym, **flags)
            elif sym not in lhs.exp.free_symbols:
                return _solve(lhs.base - rhs**(1/lhs.exp), sym, **flags)
            else:
                # sym in lhs.exp.free_symbols
                if not rhs:
                    # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
                    # the same place
                    sol_base = _solve(lhs.base, sym, **flags)
                    if not sol_base:
                        return sol_base  # no solutions to remove so return now
                    return list(ordered(set(sol_base) - set(
                    _solve(lhs.exp, sym, **flags))))
                elif rhs == 1:
                    if  lhs.base == 0:
                        return _solve(lhs.exp, sym, **flags)
                    else:
                        # 1 = foo**0, 1**foo, (-1)**even
                        b1 = _solve(lhs.base - 1, sym, **flags)
                        b_1 = _solve(lhs.base + 1, sym, **flags)
                        e0 = _solve(lhs.exp, sym, **flags)
                        sol = list(b1) + list(e0)
                        for b in b1:
                                if e.subs(sym, b).is_even is not False:
                                    sol.append(b)
                        return list(ordered(sol))
                elif lhs.base.is_positive and lhs.exp.is_real:
                    return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
```

Reply:
```for b in b1:
                                if e.subs(sym, b).is_even is not False:
                                    sol.append(b)
```
here i  had replaced ```b1``` with ```b_1``` and instead of ```eq.subs(sym, b).is_even is not False``` i think this is better ```eq.subs(sym, b).is_even is not False```.
Thanks @smichr .

Reply:
But this is still not getting solved ```x**(x+4)-4```.